user_sync/credential_manager.py

```diff
@@ -42,5 +46,22 @@ def get_credentials(credential_type, credential_id, **kwArgs):
     :type kwArgs: dict
     :rtype dict | str | None
     '''
+    if credential_type == DIRECTORY_CREDENTIAL_TYPE:
+        config = kwArgs['config']
+        if "credential_manager" in config:
+            cred_man = config['credential_manager']
+            for supported in KEYRING_SUPPORTED:
+                if cred_man['type'] == supported:
+                    return get_credentials_from_keyring(cred_man['service_name'], config['username'])
```

Comment:
same problem here as above: you can't be guaranteed that 'service_name' is a key in cred_man or that 'username' is a key in config.  So either add some 'in' tests to the if statement above or use the .get(...) form in this call.  I recommend enhancing the if above.